src/Event/AdminEventExtension.php

```diff
@@ -38,89 +38,100 @@ public function __construct(EventDispatcherInterface $eventDispatcher)
 
     public function configureFormFields(FormMapper $form)
     {
-        $this->eventDispatcher->dispatch(
-            'sonata.admin.event.configure.form',
-            new ConfigureEvent($form->getAdmin(), $form, ConfigureEvent::TYPE_FORM)
+        EventDispatcherHelper::dispatch(
```

Comment:
I don't like this helper pattern. Can't you just change the required interface? The EventDispatcherInterface component extends the contracted interface.

Reply:
> The EventDispatcherInterface component extends the contracted interface.

But only on Symfony 4.3 :wink: So there is no easy fix to support Symfony 3.4 and 4.3

https://github.com/symfony/symfony/blob/3.4/src/Symfony/Component/EventDispatcher/EventDispatcherInterface.php

Reply:
4.4 is the new LTS and v5 is out now, maybe time to drop v3 support?

Reminder: Dependency drop may be done in a minor release, this is not breaking.

Reply:
I am for dropping 3.4

Reply:
We can drop support for 3.4. It will simplify the patch a bit.

We would still need to use `LegacyEventDispatcherProxy` though since people could use a custom EventDispatcher implementation on 4.3 that has the old signature. Its deprecated but it works and needs to be supported.

Reply:
We should update the https://github.com/sonata-project/SonataAdminBundle/blob/3.x/CONTRIBUTING.md#dependency-changes, that we only support the latest LTS version

Reply:
Should be done on dev-kit

Reply:
So the fix for support only the event-dispatcher ^4.3 looks like this:

https://github.com/sonata-project/SonataAdminBundle/compare/3.x...dmaicher:fix_event_dispatcher_deprecations_drop_3.4

I did not adjust any other dependencies yet.

Is this the way we want to go?

This would mean that we only support Symfony 4.3 and 4.4 though. And 4.3 is out of maintenance soon too (even before 3.4).

Reply:
I am going to give the v3 drop a go.

Reply:
The dev-kit PR has already been done apparently: https://github.com/sonata-project/dev-kit/pull/535